src/CanvasTileRenderer.ts

```diff
@@ -25,18 +24,19 @@ export class CanvasTileRenderer
     dontUseTransform = false;
 
     /** @param renderer */
-    constructor(renderer: AbstractRenderer)
-    {
+    constructor(renderer: IRenderer) {
         this.renderer = renderer;
         this.tileAnim = [0, 0];
     }
 
     // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-    static getInstance(renderer: any): CanvasTileRenderer
-    {
-        if (!renderer.plugins.tilemap)
-        {
-            renderer.plugins.tilemap = new CanvasTileRenderer(renderer);
+    static getInstance(renderer: any): CanvasTileRenderer {
+        if (!renderer.plugins.tilemap) {
+            extensions.add({
```

Comment:
I am not sure but you are returning `render.plugins.tilemap` that is at the moment undefined. 

Will extension.add assign a new instance of tilemap in `render.plugins.tilemap` in sync? otherwise it will return undefined not CanvasTileRenderer